src/lib/utils/hooks/useTheme.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useContext } from 'react'
+
+/**
+ * TODO: Ensure the default colors here match the default colors in _variables.scss
+ */
+
+const defaultTheme: ThemeContextType = {
+  colors: {
+    success: '#32a330',
+    info: '#017fa5',
+    warning: '#cc9f00',
+    error: '#c13415',
+  },
+}
+
+export type ThemeContextType = {
+  colors: {
+    success: string
+    info: string
+    warning: string
+    error: string
+  }
+}
+
+/**
+ * This must be exported to use the context in class components.
+ */
+export const ThemeContext = React.createContext<ThemeContextType>(defaultTheme)
+
+export type ThemeProviderProps = {
+  theme?: ThemeContextType
+}
+
+export const ThemeProvider: React.FunctionComponent<ThemeProviderProps> = ({
+  theme = defaultTheme,
+  children,
+}) => {
+  return <ThemeContext.Provider value={theme}>{children}</ThemeContext.Provider>
```

Comment:
If we want to do this, we would probably want to merge the provided partial theme object with a default theme, rather than completely replace.

Reply:
I agree. We should do the deep merge to make sure we get all the props